### PR TITLE
Readme cleanup

### PR DESCRIPTION
**Pull request by jmklix**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,9 +9,8 @@ to Java by the [aws-crt-java](https://github.com/awslabs/aws-crt-java) package.
 * [Samples](samples)
 * [Getting Help](#getting-help)
 * [FAQ](./documents/FAQ.md)
-* [Giving Feedback and Contributions](#giving-feedback-and-contributions)
+* [API Docs](https://aws.github.io/aws-iot-device-sdk-java-v2/)
 * [MQTT5 User Guide](./documents/MQTT5_Userguide.md)
-* [DEVELOPING](./documents/DEVELOPING.md)
```
Comment (xiazhvera) [inline README.md:14]: Would we like to keep the development instruction here?

Comment (jmklix) [reply README.md:14]: I think it will be best down with the getting help links. What are your thoughts?

File: README.md [hunk 2]
```diff
@@ -125,18 +87,11 @@ Please make sure to check out our resources too before opening an issue:
...
-
-## Giving Feedback and Contributions
-
-We need your help in making this SDK great. Please participate in the community and contribute to this effort by submitting issues, participating in discussion forums and submitting pull requests through the following channels.
-
 * [Contributions Guidelines](./documents/CONTRIBUTING.md)
-* Articulate your feature request or upvote existing ones on our [Issues](https://github.com/aws/aws-iot-device-sdk-java-v2/issues?q=is%3Aissue+is%3Aopen+label%3Afeature-request) page.
-* Create discussion questions [here](https://github.com/aws/aws-iot-device-sdk-java-v2/discussions)
-* Find a bug open an [issue](https://github.com/aws/aws-iot-device-sdk-java-v2/issues)
+* [DEVELOPING](./documents/DEVELOPING.md)
```
Comment (sbSteveK) [inline README.md:91]: Spelling in the file at this link.

**Review by sbSteveK — APPROVED** [review]
spelling fix and good to go

File: README.md (hunk 2 above)
Comment (jmklix) [reply README.md:91]: fixed

**Pull request merged**